pkg/convert/pprof/profile.go

```diff
@@ -187,35 +190,34 @@ func (p *RawProfile) Parse(ctx context.Context, putter storage.Putter, _ storage
 		if p.SampleTypeConfig != nil {
 			sampleTypes = p.SampleTypeConfig
 		}
-		p.parser = NewParser(ParserConfig{
-			SpyName:             md.SpyName,
-			Labels:              md.Key.Labels(),
-			Putter:              putter,
-			SampleTypes:         sampleTypes,
-			SkipExemplars:       p.SkipExemplars,
-			StackFrameFormatter: StackFrameFormatterForSpyName(md.SpyName),
-		})
+
+		if p.StreamingParser {
+			p.parser = streaming.NewStreamingParser(streaming.ParserConfig{
+				SpyName:       md.SpyName,
+				Labels:        md.Key.Labels(),
+				Putter:        putter,
+				SampleTypes:   sampleTypes,
+				SkipExemplars: p.SkipExemplars,
+			})
+		} else {
+			p.parser = NewParser(ParserConfig{
+				SpyName:             md.SpyName,
+				Labels:              md.Key.Labels(),
+				Putter:              putter,
+				SampleTypes:         sampleTypes,
+				SkipExemplars:       p.SkipExemplars,
+				StackFrameFormatter: StackFrameFormatterForSpyName(md.SpyName),
+			})
+		}
 
 		if p.PreviousProfile != nil {
-			// Ignore non-cumulative samples from the PreviousProfile
-			// to avoid duplicates: although, presence of PreviousProfile
-			// tells that there are cumulative sample types, it may also
-			// include regular ones.
```

Comment:
Please leave the comment :) I fixed this place two times in past and therefore decided to put it here: it is the only case when we want to grab only samples of cumulative type (also, it would be nice to rename the argument to something like `cumulativeOnly` just for better readability)

Reply:
it is just the comment is kinda related to a boolean arg and it is just "true", it felt it is not clear what the comment is reffering to
I guess if lave the comment and do cumulativeOnly:=true/false it would be better